public/content/documentation/web/component/link.md

```diff
@@ -6,75 +6,485 @@ How to test a link
+  About
+</a>
+```
+
+<example>
+<a href="/about/">
+  About
+</a>
+</example>
+
+### Name links logically
+
+- **Do not** use a heading with a generic link below. 
+- Instead, make the heading a link or programmatically associate the link with the heading using <code>aria-describedby</code>.
+
+#### Bad example
+
+```html
+<h3>About our coffee subscriptions</h3>
+<p>Get the best coffee delivered to your door</p>
+<a href="/about/">
+   Learn more
+</div>
+```
 
-## Developer Notes
+#### Good examples
 
-### Name
+```html
+<h3><a href="/about/">About our coffee subscriptions</a></h3>
+<p>Get the best coffee delivered to your door</p>
+```
 
-- Typically doesn’t have a name or description since there must be only one instance per page.
+```html
+<h3 id="unique-id">About our coffee subscriptions</h3>
+<p>Get the best coffee delivered to your door</p>
+<a href="/about/" aria-describedby="unique-id">
+   Learn more
+</div>
+```
 
-## Videos
+### Making a link with no `href` focusable
 
-- Videos go here
-<video controls>
-  <source src="media/video/native/button/buttonIosVoiceover.webm" type="video/webm">
-  Your browser does not support the video tag.
-</video>
+- **Do not** put anything but a URI in the `href`
+- A link with no `href` will not be focusable with the keyboard without `tabindex="0"`.
+- Add `role="link"` to ensure screen reader reads the role
+
+```html
+<a tabindex="0" role="link">
+  About
+</a>
+```
+
+<example>
+<a tabindex="0" role="link">
+  About
+</a>
+</example>
+
+### Avoid custom elements
+
+This custom button requires extra attributes and keyboard event listeners.
+
+```html
+<custom-element role="link" tabindex="0">
+  About
+</custom-element>
+```
+
+### Repeating text links
+
+Sometimes the design will call for multiple links with the same text label. In a case like this, `aria-label` can be used to name each link's purpose.
+
+```html
+<button>Get free coffee</button>
+<a href="/free-coffee-tc/" aria-label="Free coffee terms and conditions">
+  Terms & Conditions
+</a>
+<button>Get free donuts</button>
+<a href="/free-donuts-tc/" aria-label="Free donuts terms and conditions">
+  Terms & Conditions
+</a>
+```
+
+### Don't duplicate the visible text name in the aria-label
+
+**Do not** repeat the inner text content of a link in the `aria-label`.
+
+#### Bad example
+```html
+<a href="/do-NOT-repeat-yourself/" 
+   aria-label="Do NOT repeat yourself">
+   Do not repeat yourself
+</div>
+```
+
+### Don't use javascript in href
+
+- **Do not** use `"href="javascript:void(0)"`. 
+- When screen readers read the href, it becomes confusing and nonsensical 
+
+#### Bad example
+```html
+<a href="javascript:void(0)">
+   Do not use javascript in href
+</div>
+```
+
+### Don't use "#" in href
+
+#### Bad example
+
+```html
+<a href="#">
+   Do not use # to populate the href
+</div>
+```
+
+## Disabled links
+
+- If it's unavoidable to have a disabled link present you'll need these attributes for the screen reader.
```

Comment:
Truly asking for simply stating what we want plainly in addition to making people review the code snippet... what if we added a sub bullet list here with the three attributes like (but with better sub listing because this box doesn't allow for the indent):

- If it’s unavoidable to have a disabled link present you’ll need these attributes for the screen reader.
- - `tabindex="0"`
- - `role="link"`
- - `aria-disabled="true"`